10up/ElasticPress
# New Sync page with advanced options

### Description of the Change
Introduces a refreshed Sync page with an Advanced Options section for setting additional parameters for a sync, bringing some of the functionality from the WP CLI command to the UI.

![elasticpress test_wp-admin_admin php_page=elasticpress-sync (1)](https://github.com/10up/ElasticPress/assets/4100733/8b7bdb7c-cf3f-4c6c-a45b-1fb568a1f6d5)

This PR includes a number things:

  - Updates the `SyncProvider` component to support additional arguments when syncing. Auto syncing and initial sync management have been removed so that they can be managed at the UI level.
  - Adds a `SettingsScreenProvider` component that sets a common layout for ElasticPress settings screens and provides a method for displaying toast/snackbar type notifications.
  - Adds a `SyncSettingsProvider` component for managing the state of selected sync settings across components.
  - Combines the separate Sync and Delete and Sync panels into a single panel with a checkbox to perform a Delete and Sync.
  - Adds the date and time to log messages.
  - Adds Clear log and Copy log to clipboard buttons.
  - Adds options for Indexables, Post Types, Object IDs, and an Offset.

The only regression against the current Sync page is that the previous sync status has been removed. The sync history will be added in a separate PR, now that #3647 has been merged.

This PR updates existing E2E tests for Syncing, but tests for new options will come separately.


Closes #3640

### How to test the Change
1. Visit _ElasticPress > Sync_.
2. It should be possible to Start, Pause, Resume, and Stop a sync as before.
3. Checking _Delete all data and start fresh sync_ should delete all data and send mapping when syncing.
4. The sync results and log should reflect the choices made in the Advanced Options section.

### Changelog Entry
Changed - Combined the separate Sync and Delete and Sync panels into a single panel with a checkbox to perform a Delete and Sync.
Added - Added the date and time to log messages on the Sync Page.
Added - Added Clear log and Copy log to clipboard buttons to the Sync page.
Added - Added Advanced Options to the Sync page for setting Content Types, Post Types, Object IDs, and an Offset for a sync.

### Credits

Props @JakePT @felipeelia Apurv Ray @brandwaffle @anjulahettige 

### Checklist:
